spec/classes/mkuser_spec.rb

```diff
@@ -1,8 +1,22 @@
 require 'spec_helper'
 
 describe 'common' do
+
+  # purge_ssh_keys was introduced with Puppet 3.6.0
+  # we need to know which version of Puppet is running this test
+  # to decide which results we need to expect
+  # dirty trick to get the running version of Puppet:
+  clientversion = `facter puppetversion`
+  # test environments contains no facts, we need to set it as fact
+
+  let(:facts) do
+    {
+      :osfamily      => 'RedHat',
+      :puppetversion => clientversion,
+    }
+  end
+
```

Comment:
This block was moved to top so these settings are available for all later tests.